src/cloud/aws.rs

```diff
@@ -0,0 +1,278 @@
+        std::env::set_var("AWS_ACCESS_KEY_ID", &self.access_key_id);
+        std::env::set_var("AWS_SECRET_ACCESS_KEY", &self.secret_access_key);
+        let region: String = self.region.clone();
+        Ok(aws_config::defaults(BehaviorVersion::latest())
+            .region(Region::new(region))
+            .http_client(hyper_client)
+            .load()
+            .await)
+    }
+
+    // This code is very unholy, but thanks to Jeff Bezos for making the AWS SDK so complicated
+    pub fn from_secret(secret: &Secret, region: String) -> color_eyre::Result<Self> {
+        let aws_access_key_id: String = secret
+            .data
+            .as_ref()
+            .and_then(
+                |f: &std::collections::BTreeMap<String, k8s_openapi::ByteString>| {
+                    f.get("AWS_ACCESS_KEY_ID")
+                },
+            )
+            .ok_or_else(|| anyhow!("AWS_ACCESS_KEY_ID not found in secret"))
+            // into String
+            .and_then(|key| String::from_utf8(key.clone().0.to_vec()).map_err(|e| e.into()))?;
+
+        let aws_secret_access_key: String = secret
+            .data
+            .as_ref()
+            .and_then(|f| f.get("AWS_SECRET_ACCESS_KEY"))
+            .ok_or_else(|| anyhow!("AWS_SECRET_ACCESS_KEY not found in secret"))
+            .and_then(|key| String::from_utf8(key.clone().0.to_vec()).map_err(|e| e.into()))?;
+
+        Ok(Self {
+            access_key_id: aws_access_key_id,
+            secret_access_key: aws_secret_access_key,
+            region,
+        })
+    }
+}
+
+#[async_trait]
+impl Provisioner for AWSProvisioner {
+    async fn create_exit_node(
+        &self,
+        auth: Secret,
+        exit_node: ExitNode,
+    ) -> color_eyre::Result<ExitNodeStatus> {
+        let provisioner = exit_node
+            .metadata
+            .annotations
+            .as_ref()
+            .and_then(|annotations| annotations.get(EXIT_NODE_PROVISIONER_LABEL))
+            .ok_or_else(|| {
+                anyhow!(
+                    "No provisioner found in annotations for exit node {}",
+                    exit_node.metadata.name.as_ref().unwrap()
+                )
+            })?;
+
+        let password = generate_password(32);
+
+        let cloud_init_config = generate_cloud_init_config(&password, CHISEL_PORT);
+        let user_data = base64::engine::general_purpose::STANDARD.encode(cloud_init_config);
+
+        let aws_api: aws_config::SdkConfig = AWSIdentity::from_secret(&auth, self.region.clone())?
+            .generate_aws_config()
+            .await?;
+
+        let ssm_client = aws_sdk_ssm::Client::new(&aws_api);
+        let parameter_response = ssm_client
+            .get_parameter()
+            .name("/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id")
+            .send()
+            .await?;
+        let ami = parameter_response.parameter.unwrap().value.unwrap();
```

Comment:
after suffering much writing this godawful AWS provider, I have learned that AWS was even worse than I originally conceived
 
I am convinced that AWS is a ploy by Amazon to torture developers and operations teams, to satisfy Bezos' personal sadistic desires

Reply:
we should probably move that magic value to a const tbh, easier to change later on??

Reply:
agreed